package/piksi_ins/overlay/usr/bin/pose_daemon_monitor

```diff
@@ -0,0 +1,64 @@
+#!/bin/sh
+
+# shellcheck disable=1091,2039
+
+. /etc/init.d/common.sh
+
+export log_tag="pose_daemon_monitor" # required by logging.sh
+. /etc/init.d/logging.sh
+
+pose_daemon_control=/etc/init.d/S96piksi_ins
+pose_daemon_state=/var/run/pose_daemon.state
+
+pose_daemon_lic_succ=license_success
+pose_daemon_lic_fail=license_failure
+
+# Daemon should finish it's license check within 6s
+pose_daemon_lic_deadline=60
+
+recheck_interval=0.1
+check_count=0
+
+# At most we'll restart PoseDaemon 4 times (30 seconds = ((4+1)*6))
+#   to attempt to get past the HW challenge.
+restart_limit=4
+restart_count=0
+
+restart_pose_daemon()
+{
+  if [[ "$restart_count" -ge "$restart_limit" ]]; then
+    logw "Restarting device, Piksi INS hardware challenge failed $restart_limit times..."
```

Comment:
should this come out as sbp via `--sbp` flag and do we need to guard a few seconds so that `reboot -f` doesn't squelch it?

Reply:
Maybe, how visible do we want this to be?

Reply:
visible in hitl results

Reply:
ok, updated